lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -1792,6 +1846,56 @@ bool HandshakeBuilder::visitHandshake(BufferOp op) {
     return false;
 }
 
+bool HandshakeBuilder::visitHandshake(ExternalMemoryOp op) {
+
+  // The external memory input is a bundle containing equivalent bundles to the
+  // remainder of inputs to this component. Due to this, we simply need to
+  // connect everything.
+
+  // Port list format:
+  // [0]: external memory bundle { like everything below, but inside a bundle }
+  // [...]: [{store data, store address}, ...]
+  // [...]: [load address, ...]
+  // [...]: [load data, ...]
+  // [...]: [control output, ...]
+  auto inBundle = op.getOperand(0).getType().dyn_cast<BundleType>();
```

Comment:
nit: i think this could just be a `cast`, which would make the assertion more clear in the case of a failed assumption